src/components/analyses/listing/Listing.js

```diff
@@ -396,7 +396,7 @@ function Listing(props) {
                         if (analysisStatus !== found.status) {
                             found.status = analysisStatus;
                             found.enddate = message.payload.enddate;
-                            setData({ analyses: [...data.analyses] });
+                            setData({ analyses: [...data.analyses], ...data });
```

Comment:
I think the copy of `...data` needs to come before the new value of `analyses`, otherwise the old values merged from `data` will clobber any new values.

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#Spread_in_object_literals

I think this is technically working now because the `found` value from state is getting mutated directly, which we shouldn't be doing.

Instead, `found` should be copied with the new values set in the copy:
```javascript
const newAnalyses = data.analyses.map((analysis) => analysis.id === message.payload.id
    ? { ...analysis, status: analysisStatus,  enddate: message.payload.enddate }
    : analysis);

setData({ ...data, analyses: newAnalyses });
```

Reply:
Makes sense. Will update it.